cpp/src/sort/rank.cu

```diff
@@ -207,8 +211,10 @@ void rank_average(rmm::device_vector<size_type> const &group_keys,
   // algorithm: reduce_by_key(dense_rank, 1, n, min_count)
   //            transform(min+(count-1)/2), scatter
   using MinCount = thrust::tuple<size_type, size_type>;
```

Comment:
```suggestion
  using MinCount = thrust::pair<size_type, size_type>;
```

Reply:
`thrust::pair` works.
Now, all can be `rmm::device_vector`. (remove device_span tie_sorted argument)
Update the lambdas `thrust::get<0>(` to `.first`, `thrust::get<1>(` to `.second`.

Reply:
Ah, that's much better. THanks!